File: grida-domains/domain-image/src/main/java/org/grida/profileimage/ProfileImageGenerator.java
package org.grida.profileimage;

import lombok.RequiredArgsConstructor;
import org.grida.image.ImageMetaData;
import org.grida.image.ImageType;
import org.grida.processor.imagegenerate.ImageGenerateProcessor;
import org.grida.processor.imagegenerate.ImageGenerateResult;
import org.grida.prompt.PromptProvider;
import org.springframework.stereotype.Component;

import static org.grida.prompt.Prompt.GENERATE_PROFILE_IMAGE;

@Component
@RequiredArgsConstructor
public class ProfileImageGenerator {

    private final PromptProvider promptProvider;
    private final ImageGenerateProcessor imageGenerateProcessor;

    public ImageMetaData generate(ProfileImageGenerateKey key) {
        String prompt = promptProvider.provide(GENERATE_PROFILE_IMAGE, key.toPromptKeyword());

        ImageGenerateResult result = imageGenerateProcessor.proceed(prompt, 1);

        return ImageMetaData.withRandomUuidFilename(
                result.getUrl(),
                ImageType.PROFILE
        );
    }
}

File: grida-core/core-domain/src/main/kotlin/org/grida/domain/profileimage/ProfileImageGenerator.kt
package org.grida.domain.profileimage

import org.grida.image.OpenAiImageClient
import org.springframework.stereotype.Component

@Component
class ProfileImageGenerator(
    private val openAiImageClient: OpenAiImageClient
) {

    fun generate(
        appearance: Appearance
    ): String {
        val prompt = ProfileImagePrompt(appearance)
        val imageUrl = openAiImageClient.generate(prompt.apply())

        return imageUrl
    }
}
